pos_order_to_sale_order/models/sale_order.py

```diff
@@ -13,6 +13,14 @@ def _prepare_from_pos(self, order_data):
         PosSession = self.env["pos.session"]
         session = PosSession.browse(order_data["pos_session_id"])
         SaleOrderLine = self.env["sale.order.line"]
+        partner_pricelist = (
+            order_data["pricelist_id"]
+            if order_data["pricelist_id"]
+            else self.env["res.partner"]
+            .browse(order_data["partner_id"])
+            .property_product_pricelist.id
+        )
+        self.pricelist_id = partner_pricelist
```

Comment:
naive question: pricelist should not be initialized by partner_id native onchange ?